COMET.Web.Common/Services/NamingConventionService/BaseNamingConventionService.cs

```diff
@@ -0,0 +1,101 @@
+
+using Microsoft.Extensions.Logging;
+
+public class BaseNamingConventionService<TEnum> : INamingConventionService<TEnum> where TEnum : Enum
+{
+    /// <summary>
+    /// <see cref="Dictionary{TKey,TValue}" /> that holds the defined naming convention
+    /// </summary>
+    private readonly Dictionary<string, string> definedNaming = new(StringComparer.OrdinalIgnoreCase);
+
+    /// <summary>
+    /// The <see cref="ILogger{TCategoryName}" />
+    /// </summary>
+    private readonly ILogger<INamingConventionService<TEnum>> logger;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="NamingConventionService" /> class.
+    /// </summary>
+    /// <param name="logger">The <see cref="ILogger{TCategoryName}" /></param>
+    protected BaseNamingConventionService(ILogger<INamingConventionService<TEnum>> logger)
+    {
+        this.logger = logger;
+    }
+
+    /// <summary>
+    /// Initializes this service
+    /// </summary>
+    /// <returns>A <see cref="Task" /></returns>
+    public async Task InitializeService()
+    {
+        var namingConvention = await this.GetNamingConventionConfiguration();
+
+        foreach (var namingConventionKind in Enum.GetValues(typeof(TEnum)))
+        {
+            if (namingConvention.TryGetValue(namingConventionKind.ToString(), out var namingConventionValue))
+            {
+                this.definedNaming[namingConventionKind.ToString()] = namingConventionValue;
+            }
+            else
+            {
+                this.logger.LogWarning("{namingConventionKind} is missing from the Naming Convention configuration file", namingConventionKind.ToString());
+            }
+        }
+    }
+
+    /// <summary>
+    /// Gets the value for naming convention
+    /// </summary>
+    /// <param name="namingConventionKey">The naming convention key</param>
+    /// <returns>The defined naming convention, if exists</returns>
+    public string GetNamingConventionValue(string namingConventionKey)
+    {
+        return this.definedNaming.TryGetValue(namingConventionKey, out var namingConventionValue) ? namingConventionValue : string.Empty;
+    }
+
+    /// <summary>
+    /// Gets the value for naming convention
+    /// </summary>
+    /// <param name="namingConventionKind">The <see cref="NamingConventionKind" /></param>
+    /// <returns>The defined naming convention, if exists</returns>
+    public string GetNamingConventionValue(TEnum namingConventionKind)
+    {
+        return this.GetNamingConventionValue(namingConventionKind.ToString());
+    }
+
+    /// <summary>
+    /// Gets the naming convention configuration
+    /// </summary>
+    /// <returns>A <see cref="IReadOnlyDictionary{TKey,TValue}"/> of the naming convention configuration</returns>
+    /// <exception cref="NotImplementedException"></exception>
+    public virtual Task<IReadOnlyDictionary<string, string>> GetNamingConventionConfiguration()
+    {
+        throw new NotImplementedException();
```

Comment:
why throwing exception there? You could just return the definedNaming dictionary ?

Reply:
That would be a bit irrelevant for this method as that collection hasn't been filled up - the goal would be for any class that inherit this one would have to override this method themselves, depending on how they want to retrieve the configuration.

I could, however, just return an empty dictionary, if you prefer it like that

Reply:
Understood, took me a bit time to follow your implementation. My recommendation for this case : put the Base class as abstract and set that method abstract also. It will force any subclass to implement it (which is what we want). That base class will never be used as a concrete class anyway